### PR TITLE
URL Page Routing

### PR DESCRIPTION
**Pull request by jacob6838**
Adding url-based routing to the CVManager
- Created sub-paths for admin pages, sub-pages, and add/edit sub-sub-pages
- Added 404 page for invalid routes
- Embedded emails/IPs for admin editing in URLs

### CODE REVIEW

File: webapp/src/pages/Map.tsx [hunk 1]
```diff
@@ -620,6 +620,7 @@ function MapPage(props: MapPageProps) {
   }
 
   return (
+    // <>
```
Comment (payneBrandon) [inline webapp/src/pages/Map.tsx]: Needed?

File: webapp/src/pages/404.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      </Link>
+    </div>
+  )
+}
+
+export default function NotFound() {
+  const dispatch: ThunkDispatch<RootState, void, AnyAction> = useDispatch()
+  return (
+    <div>
+      <h1>Oops! You seem to be lost.</h1>
```
Comment (payneBrandon) [inline webapp/src/pages/404.tsx]: haha nice touch

File: webapp/src/features/adminEditRsu/AdminEditRsu.tsx [hunk 3]
```diff
@@ -112,11 +109,11 @@ const AdminEditRsu = (props: AdminEditRsuProps) => {
     },
   })
 
-  const { rsuData } = props
+  const { rsuIp } = useParams<{ rsuIp: string }>()
 
   useEffect(() => {
-    dispatch(getRsuInfo(rsuData.ip))
-  }, [dispatch, rsuData.ip])
+    dispatch(getRsuInfo(rsuIp))
```
Comment (payneBrandon) [inline webapp/src/features/adminEditRsu/AdminEditRsu.tsx]: With the urlRouting, this allows invalid rsuIp to be "edited". We'll want to add a check for invalid edit attempts, as well as trying to directly edit ones they don't have permissions for.

The backing admin-rsu function is set to False on organization_required which I believe would mean a user could bypass our security here and edit an RSU by IP that they shouldn't be able to

File: webapp/src/features/adminEditUser/AdminEditUser.tsx [hunk 4]
```diff
@@ -62,11 +58,11 @@ const AdminEditUser = (props: AdminEditUserProps) => {
     },
   })
 
-  const { userData } = props
+  const { email } = useParams<{ email: string }>()
 
   useEffect(() => {
-    dispatch(getUserData(userData.email))
-  }, [userData, dispatch])
+    dispatch(getUserData(email))
```
Comment (payneBrandon) [inline webapp/src/features/adminEditUser/AdminEditUser.tsx]: We'll want a check for invalid edit attempts (non-existent users)

File: webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx [hunk 5]
```diff
@@ -35,6 +37,12 @@ const AdminEditOrganization = () => {
     },
   })
 
+  const { orgName } = useParams<{ orgName: string }>()
+
+  useEffect(() => {
+    getOrgData({ orgName })
```
Comment (payneBrandon) [inline webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx]: We'll want a check for editing non-existent orgs. This edit page also doesn't seem to work, I'm unable to edit an org name.

**Review by payneBrandon — APPROVED** [review]
Just a couple comments for scenario oddities. Overall the url navigation is looking slick!

**Review by payneBrandon — CHANGES_REQUESTED** [review]
Just a couple comments for scenario oddities. Overall the url navigation is looking slick!

File: webapp/src/pages/Map.tsx (hunk 1 above)
Comment (jacob6838) [reply webapp/src/pages/Map.tsx]: Removed!

File: webapp/src/features/adminEditRsu/AdminEditRsu.tsx (hunk 3 above)
Comment (jacob6838) [reply webapp/src/features/adminEditRsu/AdminEditRsu.tsx]: Added a check!

File: webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx (hunk 5 above)
Comment (jacob6838) [reply webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx]: Added a check!

Comment (jacob6838) [reply webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx]: Fixed the issue of orgs not re-naming as well

File: webapp/src/features/adminEditUser/AdminEditUser.tsx (hunk 4 above)
Comment (jacob6838) [reply webapp/src/features/adminEditUser/AdminEditUser.tsx]: Added a check!

**Comment by jacob6838** [comment]
I did a bunch of testing, and the changes look good. Each of the edit pages are now protected from invalid values

File: webapp/src/features/adminEditRsu/AdminEditRsu.tsx [hunk 6]
```diff
@@ -361,6 +372,11 @@ const AdminEditRsu = () => {
             </button>
           </div>
         </Form>
+      ) : (
+        <Typography variant={'h4'} style={{ color: '#fff' }}>
+          Unknown RSU IP address. Either this RSU does not exist, or you do not have access to it.{' '}
+          <Link to="dashboard/admin/rsus">RSUs</Link>
```
Comment (payneBrandon) [inline webapp/src/features/adminEditRsu/AdminEditRsu.tsx]: Clicking this link gets us to an error page: 
image

**Review by payneBrandon — CHANGES_REQUESTED** [review]
I'm not seeing the test@gmail.com user when I go to review users. Not sure if this is intended since I logged in with this user? Manually going to the edit page for this user also fails. 
image

File: webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx [hunk 7]
```diff
@@ -35,6 +47,23 @@ const AdminEditOrganization = () => {
     },
   })
 
+  const { orgName } = useParams<{ orgName: string }>()
+
+  useEffect(() => {
+    if (
+      (orgData ?? []).find((organization: AdminOrgSummary) => organization.name === orgName) &&
```
Comment (payneBrandon) [inline webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx]: Seeing an error on loading this page: 
image

Comment (jacob6838) [reply webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx]: this one is weird, there is a null organization?? This is inside the find method... Added the check, but still kinda confused

File: webapp/src/features/adminEditRsu/AdminEditRsu.tsx (hunk 6 above)
Comment (jacob6838) [reply webapp/src/features/adminEditRsu/AdminEditRsu.tsx]: Yup, I changed the links to be absolute and didn't put the starting slash... fixed...

**Comment by jacob6838** [comment]
> I'm not seeing the [test@gmail.com](mailto:test@gmail.com) user when I go to review users. Not sure if this is intended since I logged in with this user? Manually going to the edit page for this user also fails. image

I am not able to reproduce this. I see the test@gmail.com user in the users page... Thinking this may be an issue with the api not returning the right data?

**Review by drewjj — APPROVED** [review]
This seems to work well for me. Why do we have a RSU Map page if it isn't used though?

File: webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx (hunk 7 above)
Comment (payneBrandon) [reply webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx]: Looks like the org list isn't getting populated maybe: 
image

**Comment by payneBrandon** [comment]
> This seems to work well for me. Why do we have a RSU Map page if it isn't used though?

Agreed, can we drop the RSU Map page?

File: webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx (hunk 7 above)
Comment (jacob6838) [reply webapp/src/features/adminEditOrganization/AdminEditOrganization.tsx]: This is resolved!

**Comment by jacob6838** [comment]
@drewjj The RSU map page has been removed, and I fixed the final few bugs on the edit organization page. To my knowledge, that resolves all known issues!

**Review by drewjj — APPROVED** [review]
Looks good!

**Review by payneBrandon — APPROVED** [review]
lgtm!

**Pull request merged**